tests/integration/pilot/common/routing.go

```diff
@@ -520,10 +524,12 @@ spec:
 							Count:    1,
 							Check: check.And(
 								check.OK(),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Allow-Origin", "cors.com")),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Allow-Methods", "POST,GET")),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Allow-Headers", "X-Foo-Bar,X-Foo-Baz")),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Max-Age", "86400"))),
+								check.ResponseHeaders(map[string]string{
```

Comment:
nit: slightly prefer taking in 2 strings as input. This is likely one of the only cases we will want to check multiple headers and can just call the Response Header checker multiple times

Reply:
Ok actually there was one other usage it looks like. Still my preference but I'm fine either way

Reply:
I've added both and one calls the other.